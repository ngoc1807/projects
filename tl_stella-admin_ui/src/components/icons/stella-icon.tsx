import { SvgIcon, SvgIconProps, useTheme } from "@mui/material";

interface StellaIconProps extends SvgIconProps {
  variant?: "long" | "short" | "medium";
  color1?: string;
  color2?: string;
  size?: number;
}

export function StellaIcon({
  variant,
  color1,
  color2,
  size,
  ...props
}: StellaIconProps) {
  const theme = useTheme();
  if (!color1) color1 = theme.palette.secondary.main;
  if (!color2) color2 = theme.palette.primary.main;

  switch (variant) {
    case "long":
      return (
        <SvgIcon
          color="inherit"
          viewBox="0 0 1317 105"
          titleAccess="STELLA Automotive AI"
          sx={{
            width: `${(size || 40) * (1317 / 105)}px`,
            height: `${size || 40}px`,
          }}
          {...props}
        >
          <path
            d="M583.82 100C580.54 99.9764 577.4 98.6621 575.081 96.3415C572.763 94.0208 571.451 90.8805 571.43 87.6V35.78H583.2V85C583.2 86.67 583.3 87.61 583.51 87.81C583.72 88.01 584.65 88.13 586.32 88.13H620.66C621.592 88.2346 622.536 88.1251 623.42 87.81C623.66 87.61 623.78 86.67 623.78 85V35.78H635.64V87.6C635.614 90.8806 634.299 94.0194 631.979 96.3392C629.659 98.659 626.521 99.9739 623.24 100H583.82Z"
            fill={color1}
          />
          <path
            d="M669.53 100V47.64H643.31V35.78H707.52V47.64H681.39V100H669.53Z"
            fill={color1}
          />
          <path
            d="M727.5 100C724.22 99.9764 721.08 98.6621 718.761 96.3415C716.443 94.0208 715.131 90.8805 715.11 87.6V48.18C715.131 44.8995 716.443 41.7592 718.761 39.4385C721.08 37.1179 724.22 35.8036 727.5 35.78H766.92C770.201 35.8061 773.339 37.121 775.659 39.4408C777.979 41.7606 779.294 44.8994 779.32 48.18V87.6C779.294 90.8806 777.979 94.0194 775.659 96.3392C773.339 98.659 770.201 99.9739 766.92 100H727.5ZM730 88.14H764.34C765.272 88.2446 766.216 88.1351 767.1 87.82C767.34 87.62 767.46 86.68 767.46 85.01V50.77C767.46 49.1 767.34 48.17 767.1 47.96C766.216 47.6449 765.272 47.5355 764.34 47.64H730C728.33 47.64 727.4 47.75 727.19 47.96C726.98 48.17 726.88 49.1 726.88 50.77V85C726.88 86.67 726.98 87.61 727.19 87.81C727.4 88.01 728.33 88.14 730 88.14Z"
            fill={color1}
          />
          <path
            d="M790.2 100V35.78H803.13L826.41 63.52L849.6 35.78H862.6V100H850.74V52.91L826.39 81.91L802 53V100H790.2Z"
            fill={color1}
          />
          <path
            d="M886.25 100C882.97 99.9764 879.83 98.6621 877.511 96.3415C875.193 94.0208 873.881 90.8805 873.86 87.6V48.18C873.881 44.8995 875.193 41.7592 877.511 39.4385C879.83 37.1179 882.97 35.8036 886.25 35.78H925.68C928.961 35.8061 932.099 37.121 934.419 39.4408C936.739 41.7606 938.054 44.8994 938.08 48.18V87.6C938.054 90.8806 936.739 94.0194 934.419 96.3392C932.099 98.659 928.961 99.9739 925.68 100H886.25ZM888.75 88.14H923.09C924.026 88.243 924.972 88.1336 925.86 87.82C926.1 87.62 926.21 86.68 926.21 85.01V50.77C926.21 49.1 926.1 48.17 925.86 47.96C924.972 47.6464 924.026 47.5371 923.09 47.64H888.75C887.09 47.64 886.15 47.75 885.94 47.96C885.73 48.17 885.63 49.1 885.63 50.77V85C885.63 86.67 885.73 87.61 885.94 87.81C886.15 88.01 887.09 88.14 888.75 88.14Z"
            fill={color1}
          />
          <path
            d="M972 100V47.64H945.78V35.78H1010V47.64H983.87V100H972Z"
            fill={color1}
          />
          <path d="M1017.36 100V35.78H1029.05V100H1017.36Z" fill={color1} />
          <path d="M1304.36 100V35.78H1316.05V100H1304.36Z" fill={color1} />
          <path
            d="M567.43 100H553.7L527.65 54.64L501.43 100H487.78L527.65 31.28L567.43 100Z"
            fill={color1}
          />
          <path
            d="M1297.43 100H1283.7L1257.7 54.64L1231.43 100H1217.78L1257.65 31.28L1297.43 100Z"
            fill={color1}
          />
          <path
            d="M1035.81 35.78H1049.54L1075.54 81.14L1101.76 35.78H1115.41L1075.54 104.5L1035.81 35.78Z"
            fill={color1}
          />
          <path
            d="M1122.24 100V35.78H1181.37V47.64H1134.1V61.92H1172.1V73.87H1134.1V88.14H1181.37V100H1122.24Z"
            fill={color1}
          />
          <path
            d="M53.87 44.86H101.03V36.1L53.87 36.21C49.0305 36.2179 44.3917 38.1446 40.9706 41.5676C37.5494 44.9906 35.6253 49.6305 35.62 54.47V68.13H83C85.4896 68.1327 87.8764 69.1228 89.6368 70.8832C91.3972 72.6436 92.3874 75.0304 92.39 77.52V91.35H35.62V100H101V77.5C100.992 72.7221 99.0954 68.141 95.7235 64.756C92.3515 61.3709 87.7779 59.4564 83 59.43H44.29V54.43C44.2979 51.8926 45.3101 49.4615 47.1053 47.6682C48.9005 45.8749 51.3326 44.8653 53.87 44.86V44.86Z"
            fill={color2}
          />
          <path
            d="M109.76 36.1V44.84H137.28V100H146.08V44.84H173.52V36.1H109.76Z"
            fill={color2}
          />
          <path
            d="M258.13 91.2V36.16H249.33V100H313.09V91.2H258.13Z"
            fill={color2}
          />
          <path
            d="M418.56 31.36L384.1 91.2H330.62V36.16H321.82V100H379.03H385.58H389.19L418.6 48.9L448.32 100.09H458.49L418.56 31.36Z"
            fill={color2}
          />
          <path
            d="M240.59 44.84V36.1H182.26V100H240.59V91.2H191.06V68H221.76V59.3H191.06V44.84H240.59Z"
            fill={color2}
          />
          <path
            d="M26.79 0.5C26.79 7.60976 23.9691 14.429 18.9464 19.4611C13.9238 24.4931 7.10974 27.3268 0 27.34C7.11841 27.34 13.9453 30.1678 18.9787 35.2013C24.0122 40.2347 26.84 47.0616 26.84 54.18C26.84 47.0616 29.6678 40.2347 34.7013 35.2013C39.7347 30.1678 46.5616 27.34 53.68 27.34C46.5616 27.34 39.7347 24.5122 34.7013 19.4788C29.6678 14.4453 26.84 7.61841 26.84 0.5H26.79Z"
            fill={color1}
          />
          <path
            d="M438.63 47.28V38.49H433.57V36.1H445.65V38.49H441.16V47.28H438.63Z"
            fill={color2}
          />
          <path
            d="M445.61 47.28V36.1H449.26L451.97 40.73L454.66 36.1H458.3V47.28H455.92V38L452 44.8L448 38V47.24L445.61 47.28Z"
            fill={color2}
          />
        </SvgIcon>
      );
      break;
    case "medium":
      return (
        <SvgIcon
          color="inherit"
          viewBox="0 0 686 226"
          titleAccess="STELLA Retail Automotive Edition"
          sx={{
            width: `${(size || 40) * (686 / 226)}px`,
            height: `${size || 40}px`,
          }}
          {...props}
        >
          <path
            d="M126.59 222.45C124.09 222.429 121.699 221.427 119.931 219.659C118.163 217.891 117.161 215.5 117.14 213V173.48H126.14V211C126.049 211.726 126.131 212.462 126.38 213.15C127.065 213.394 127.798 213.476 128.52 213.39H154.71C155.419 213.469 156.137 213.387 156.81 213.15C157 212.99 157.09 212.28 157.09 211V173.48H166.09V213C166.069 215.5 165.067 217.891 163.299 219.659C161.531 221.427 159.14 222.429 156.64 222.45H126.59Z"
            fill={color1}
          />
          <path
            d="M192 222.45V182.53H172V173.53H221V182.53H201V222.45H192Z"
            fill={color1}
          />
          <path
            d="M236.16 222.45C233.66 222.429 231.269 221.427 229.501 219.659C227.733 217.891 226.731 215.5 226.71 213V182.93C226.731 180.43 227.733 178.039 229.501 176.271C231.269 174.503 233.66 173.501 236.16 173.48H266.23C268.73 173.501 271.121 174.503 272.889 176.271C274.657 178.039 275.659 180.43 275.68 182.93V213C275.659 215.5 274.657 217.891 272.889 219.659C271.121 221.427 268.73 222.429 266.23 222.45H236.16ZM238.07 213.45H264.26C264.972 213.527 265.693 213.445 266.37 213.21C266.55 213.05 266.64 212.34 266.64 211.06V184.91C266.64 183.64 266.55 182.91 266.37 182.76C265.692 182.53 264.972 182.452 264.26 182.53H238.07C237.739 182.315 237.345 182.22 236.953 182.26C236.56 182.301 236.194 182.476 235.915 182.755C235.636 183.034 235.461 183.4 235.42 183.793C235.38 184.185 235.475 184.579 235.69 184.91V211C235.481 211.331 235.391 211.724 235.434 212.113C235.477 212.503 235.651 212.866 235.927 213.144C236.204 213.421 236.566 213.597 236.956 213.642C237.345 213.686 237.738 213.598 238.07 213.39V213.45Z"
            fill={color1}
          />
          <path
            d="M284 222.45V173.45H293.86L311.61 194.6L329.3 173.45H339.23V222.45H330.18V186.54L311.61 208.64L292.98 186.64V222.48L284 222.45Z"
            fill={color1}
          />
          <path
            d="M357.23 222.45C354.729 222.434 352.334 221.434 350.565 219.665C348.796 217.896 347.796 215.501 347.78 213V182.93C347.796 180.429 348.796 178.034 350.565 176.265C352.334 174.496 354.729 173.496 357.23 173.48H387.29C389.792 173.498 392.186 174.499 393.956 176.267C395.725 178.035 396.729 180.428 396.75 182.93V213C396.729 215.502 395.725 217.895 393.956 219.663C392.186 221.431 389.792 222.432 387.29 222.45H357.23ZM359.13 213.45H385.32C386.032 213.53 386.754 213.448 387.43 213.21C387.61 213.05 387.7 212.34 387.7 211.06V184.91C387.7 183.64 387.61 182.91 387.43 182.76C386.752 182.528 386.032 182.45 385.32 182.53H359.13C358.799 182.32 358.407 182.229 358.018 182.273C357.629 182.316 357.266 182.491 356.99 182.769C356.713 183.046 356.54 183.41 356.498 183.799C356.456 184.188 356.549 184.58 356.76 184.91V211C356.674 211.725 356.753 212.46 356.99 213.15C357.674 213.398 358.406 213.487 359.13 213.41V213.45Z"
            fill={color1}
          />
          <path
            d="M422.59 222.45V182.53H402.59V173.53H451.59V182.53H431.64V222.45H422.59Z"
            fill={color1}
          />
          <path d="M457.21 222.45V173.45H466.12V222.45H457.21Z" fill={color1} />
          <path d="M676.08 222.45V173.45H685V222.45H676.08Z" fill={color1} />
          <path
            d="M114.1 222.45H103.62L83.76 187.86L63.76 222.45H53.35L83.76 170.05L114.1 222.45Z"
            fill={color1}
          />
          <path
            d="M670.8 222.45H660.32L640.46 187.86L620.46 222.45H610.05L640.46 170.05L670.8 222.45Z"
            fill={color1}
          />
          <path
            d="M471.28 173.48H481.75L501.62 208.07L521.62 173.48H532L501.6 225.88L471.28 173.48Z"
            fill={color1}
          />
          <path
            d="M537.19 222.45V173.45H582.29V182.45H546.29V193.33H575.29V202.44H546.29V213.33H582.29V222.33L537.19 222.45Z"
            fill={color1}
          />
          <path
            d="M80.83 66.43L151.26 66.38V53.3L80.83 53.47C73.6 53.4779 66.6684 56.3536 61.556 61.466C56.4436 66.5784 53.5679 73.51 53.56 80.74V101.14H124.28C127.993 101.14 131.554 102.615 134.18 105.241C136.805 107.866 138.28 111.427 138.28 115.14V135.79H53.56V148.79H151.26V115.17C151.26 108.009 148.415 101.142 143.352 96.0781C138.288 91.0146 131.421 88.17 124.26 88.17H66.52V80.72C66.528 76.929 68.0388 73.2959 70.7213 70.6172C73.4038 67.9384 77.039 66.4326 80.83 66.43V66.43Z"
            fill={color2}
          />
          <path
            d="M164.32 53.33V66.38H205.42V148.78H218.57V66.38H259.54V53.33H164.32Z"
            fill={color2}
          />
          <path
            d="M385.93 135.64V53.42H372.78V148.78H468.01V135.64H385.93Z"
            fill={color2}
          />
          <path
            d="M625.56 46.25L574.09 135.64H494.21V53.42H481.06V148.78H566.52H576.29H581.69L625.62 72.46L670 148.91H685.2L625.56 46.25Z"
            fill={color2}
          />
          <path
            d="M359.73 66.38V53.33H272.6V148.78H359.73V135.64H285.74V100.98H331.6V87.99H285.74V66.38H359.73Z"
            fill={color2}
          />
          <path
            d="M40.38 0.160004C40.38 5.4247 39.343 10.6378 37.3283 15.5018C35.3136 20.3657 32.3606 24.7852 28.6379 28.5079C24.9152 32.2306 20.4957 35.1836 15.6318 37.1983C10.7678 39.213 5.5547 40.25 0.290009 40.25C5.5547 40.25 10.7678 41.287 15.6318 43.3017C20.4957 45.3164 24.9152 48.2694 28.6379 51.9921C32.3606 55.7148 35.3136 60.1343 37.3283 64.9982C39.343 69.8622 40.38 75.0753 40.38 80.34C40.38 69.7075 44.6037 59.5104 52.1221 51.9921C59.6404 44.4738 69.8374 40.25 80.47 40.25C69.8374 40.25 59.6404 36.0262 52.1221 28.5079C44.6037 20.9896 40.38 10.7925 40.38 0.160004V0.160004Z"
            fill={color1}
          />
          <path
            d="M655.54 70V56.91H648V53.33H666V56.91H659.29V70H655.54Z"
            fill={color2}
          />
          <path
            d="M666 70V53.3H671.45L675.51 60.22L679.51 53.3H684.94V70H681.4V56.21L675.5 66.32L669.57 56.24V70H666Z"
            fill={color2}
          />
        </SvgIcon>
      );
      break;
    default:
      return (
        <SvgIcon
          viewBox="0 0 866 150"
          titleAccess="STELLA"
          sx={{
            width: `${(size || 40) * (866 / 150)}px`,
            height: `${size || 40}px`,
          }}
          {...props}
        >
          <path d="M847.59 149.91V53.1H865.21V149.91H847.59Z" fill={color1} />
          <path
            d="M837.15 149.91H816.44L777.17 81.54L737.64 149.91H717.07L777.18 46.32L837.15 149.91Z"
            fill={color1}
          />
          <path
            d="M81.38 67H151.86V53.89L81.38 54.06C74.1473 54.0679 67.2131 56.9446 62.0989 62.0589C56.9846 67.1732 54.1079 74.1073 54.1 81.34V101.76H124.86C128.573 101.76 132.134 103.235 134.76 105.861C137.385 108.486 138.86 112.047 138.86 115.76V136.43H54.1V149.43H151.86V115.8C151.86 108.639 149.015 101.772 143.952 96.7081C138.888 91.6446 132.021 88.8 124.86 88.8H67.06V81.34C67.06 77.5403 68.568 73.8959 71.253 71.2072C73.9379 68.5185 77.5803 67.0053 81.38 67V67Z"
            fill={color2}
          />
          <path
            d="M164.92 53.92V66.98H206.05V149.43H219.21V66.98H260.21V53.92H164.92Z"
            fill={color2}
          />
          <path
            d="M386.68 136.28V54.01H373.52V149.43H468.81V136.28H386.68Z"
            fill={color2}
          />
          <path
            d="M626.46 46.84L574.96 136.28H495.03V54.01H481.87V149.43H567.38H577.16H582.56L626.52 73.06L670.94 149.56H686.15L626.46 46.84Z"
            fill={color2}
          />
          <path
            d="M360.46 66.98V53.92H273.27V149.43H360.46V136.28H286.43V101.6H332.31V88.6H286.43V66.98H360.46Z"
            fill={color2}
          />
          <path
            d="M40.91 0.709991C40.91 11.3505 36.6831 21.5552 29.1591 29.0791C21.6352 36.6031 11.4305 40.83 0.790009 40.83C6.06285 40.83 11.284 41.8694 16.1548 43.8887C21.0257 45.9081 25.4507 48.8678 29.1768 52.5985C32.903 56.3293 35.8571 60.758 37.8704 65.6314C39.8837 70.5047 40.9166 75.7272 40.91 81C40.9021 75.7293 41.9331 70.5087 43.944 65.6367C45.955 60.7648 48.9064 56.3369 52.6296 52.6062C56.3529 48.8756 60.7748 45.9153 65.6428 43.8947C70.5108 41.874 75.7293 40.8326 81 40.83C75.7318 40.83 70.5153 39.7922 65.6483 37.7758C60.7812 35.7595 56.3591 32.8041 52.6344 29.0785C48.9097 25.3529 45.9554 20.93 43.9403 16.0625C41.9252 11.195 40.8887 5.97815 40.89 0.709991H40.91Z"
            fill={color1}
          />
          <path
            d="M656.46 70.64V57.5H648.89L648.84 53.92H666.89V57.5H660.18V70.64H656.46Z"
            fill={color2}
          />
          <path
            d="M666.89 70.64V53.92H672.35L676.4 60.84L680.4 53.92H685.83V70.64H682.29V56.8L676.38 66.92L670.45 56.83V70.64H666.89Z"
            fill={color2}
          />
        </SvgIcon>
      );
  }
}
