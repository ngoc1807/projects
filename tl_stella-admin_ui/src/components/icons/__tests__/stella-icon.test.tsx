import { cleanup, render } from "@testing-library/react";
import { StellaIcon } from "../stella-icon";

describe("stella-icon", () => {
  afterEach(() => {
    cleanup();
  });
  it("should render correctly when no props are passed", () => {
    const { container } = render(<StellaIcon />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it("should render correctly when props are passed", () => {
    const { container } = render(
      <StellaIcon variant="long" size={24} color1="white" color2="red" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
