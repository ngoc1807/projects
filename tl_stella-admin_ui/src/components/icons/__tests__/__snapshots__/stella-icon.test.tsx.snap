// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stella-icon should render correctly when no props are passed 1`] = `
<svg
  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-iz4z57-MuiSvgIcon-root"
  focusable="false"
  role="img"
  viewBox="0 0 866 150"
>
  <path
    d="M847.59 149.91V53.1H865.21V149.91H847.59Z"
    fill="#9c27b0"
  />
  <path
    d="M837.15 149.91H816.44L777.17 81.54L737.64 149.91H717.07L777.18 46.32L837.15 149.91Z"
    fill="#9c27b0"
  />
  <path
    d="M81.38 67H151.86V53.89L81.38 54.06C74.1473 54.0679 67.2131 56.9446 62.0989 62.0589C56.9846 67.1732 54.1079 74.1073 54.1 81.34V101.76H124.86C128.573 101.76 132.134 103.235 134.76 105.861C137.385 108.486 138.86 112.047 138.86 115.76V136.43H54.1V149.43H151.86V115.8C151.86 108.639 149.015 101.772 143.952 96.7081C138.888 91.6446 132.021 88.8 124.86 88.8H67.06V81.34C67.06 77.5403 68.568 73.8959 71.253 71.2072C73.9379 68.5185 77.5803 67.0053 81.38 67V67Z"
    fill="#1976d2"
  />
  <path
    d="M164.92 53.92V66.98H206.05V149.43H219.21V66.98H260.21V53.92H164.92Z"
    fill="#1976d2"
  />
  <path
    d="M386.68 136.28V54.01H373.52V149.43H468.81V136.28H386.68Z"
    fill="#1976d2"
  />
  <path
    d="M626.46 46.84L574.96 136.28H495.03V54.01H481.87V149.43H567.38H577.16H582.56L626.52 73.06L670.94 149.56H686.15L626.46 46.84Z"
    fill="#1976d2"
  />
  <path
    d="M360.46 66.98V53.92H273.27V149.43H360.46V136.28H286.43V101.6H332.31V88.6H286.43V66.98H360.46Z"
    fill="#1976d2"
  />
  <path
    d="M40.91 0.709991C40.91 11.3505 36.6831 21.5552 29.1591 29.0791C21.6352 36.6031 11.4305 40.83 0.790009 40.83C6.06285 40.83 11.284 41.8694 16.1548 43.8887C21.0257 45.9081 25.4507 48.8678 29.1768 52.5985C32.903 56.3293 35.8571 60.758 37.8704 65.6314C39.8837 70.5047 40.9166 75.7272 40.91 81C40.9021 75.7293 41.9331 70.5087 43.944 65.6367C45.955 60.7648 48.9064 56.3369 52.6296 52.6062C56.3529 48.8756 60.7748 45.9153 65.6428 43.8947C70.5108 41.874 75.7293 40.8326 81 40.83C75.7318 40.83 70.5153 39.7922 65.6483 37.7758C60.7812 35.7595 56.3591 32.8041 52.6344 29.0785C48.9097 25.3529 45.9554 20.93 43.9403 16.0625C41.9252 11.195 40.8887 5.97815 40.89 0.709991H40.91Z"
    fill="#9c27b0"
  />
  <path
    d="M656.46 70.64V57.5H648.89L648.84 53.92H666.89V57.5H660.18V70.64H656.46Z"
    fill="#1976d2"
  />
  <path
    d="M666.89 70.64V53.92H672.35L676.4 60.84L680.4 53.92H685.83V70.64H682.29V56.8L676.38 66.92L670.45 56.83V70.64H666.89Z"
    fill="#1976d2"
  />
  <title>
    STELLA
  </title>
</svg>
`;

exports[`stella-icon should render correctly when props are passed 1`] = `
<svg
  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-3g5fjw-MuiSvgIcon-root"
  focusable="false"
  role="img"
  viewBox="0 0 1317 105"
>
  <path
    d="M583.82 100C580.54 99.9764 577.4 98.6621 575.081 96.3415C572.763 94.0208 571.451 90.8805 571.43 87.6V35.78H583.2V85C583.2 86.67 583.3 87.61 583.51 87.81C583.72 88.01 584.65 88.13 586.32 88.13H620.66C621.592 88.2346 622.536 88.1251 623.42 87.81C623.66 87.61 623.78 86.67 623.78 85V35.78H635.64V87.6C635.614 90.8806 634.299 94.0194 631.979 96.3392C629.659 98.659 626.521 99.9739 623.24 100H583.82Z"
    fill="white"
  />
  <path
    d="M669.53 100V47.64H643.31V35.78H707.52V47.64H681.39V100H669.53Z"
    fill="white"
  />
  <path
    d="M727.5 100C724.22 99.9764 721.08 98.6621 718.761 96.3415C716.443 94.0208 715.131 90.8805 715.11 87.6V48.18C715.131 44.8995 716.443 41.7592 718.761 39.4385C721.08 37.1179 724.22 35.8036 727.5 35.78H766.92C770.201 35.8061 773.339 37.121 775.659 39.4408C777.979 41.7606 779.294 44.8994 779.32 48.18V87.6C779.294 90.8806 777.979 94.0194 775.659 96.3392C773.339 98.659 770.201 99.9739 766.92 100H727.5ZM730 88.14H764.34C765.272 88.2446 766.216 88.1351 767.1 87.82C767.34 87.62 767.46 86.68 767.46 85.01V50.77C767.46 49.1 767.34 48.17 767.1 47.96C766.216 47.6449 765.272 47.5355 764.34 47.64H730C728.33 47.64 727.4 47.75 727.19 47.96C726.98 48.17 726.88 49.1 726.88 50.77V85C726.88 86.67 726.98 87.61 727.19 87.81C727.4 88.01 728.33 88.14 730 88.14Z"
    fill="white"
  />
  <path
    d="M790.2 100V35.78H803.13L826.41 63.52L849.6 35.78H862.6V100H850.74V52.91L826.39 81.91L802 53V100H790.2Z"
    fill="white"
  />
  <path
    d="M886.25 100C882.97 99.9764 879.83 98.6621 877.511 96.3415C875.193 94.0208 873.881 90.8805 873.86 87.6V48.18C873.881 44.8995 875.193 41.7592 877.511 39.4385C879.83 37.1179 882.97 35.8036 886.25 35.78H925.68C928.961 35.8061 932.099 37.121 934.419 39.4408C936.739 41.7606 938.054 44.8994 938.08 48.18V87.6C938.054 90.8806 936.739 94.0194 934.419 96.3392C932.099 98.659 928.961 99.9739 925.68 100H886.25ZM888.75 88.14H923.09C924.026 88.243 924.972 88.1336 925.86 87.82C926.1 87.62 926.21 86.68 926.21 85.01V50.77C926.21 49.1 926.1 48.17 925.86 47.96C924.972 47.6464 924.026 47.5371 923.09 47.64H888.75C887.09 47.64 886.15 47.75 885.94 47.96C885.73 48.17 885.63 49.1 885.63 50.77V85C885.63 86.67 885.73 87.61 885.94 87.81C886.15 88.01 887.09 88.14 888.75 88.14Z"
    fill="white"
  />
  <path
    d="M972 100V47.64H945.78V35.78H1010V47.64H983.87V100H972Z"
    fill="white"
  />
  <path
    d="M1017.36 100V35.78H1029.05V100H1017.36Z"
    fill="white"
  />
  <path
    d="M1304.36 100V35.78H1316.05V100H1304.36Z"
    fill="white"
  />
  <path
    d="M567.43 100H553.7L527.65 54.64L501.43 100H487.78L527.65 31.28L567.43 100Z"
    fill="white"
  />
  <path
    d="M1297.43 100H1283.7L1257.7 54.64L1231.43 100H1217.78L1257.65 31.28L1297.43 100Z"
    fill="white"
  />
  <path
    d="M1035.81 35.78H1049.54L1075.54 81.14L1101.76 35.78H1115.41L1075.54 104.5L1035.81 35.78Z"
    fill="white"
  />
  <path
    d="M1122.24 100V35.78H1181.37V47.64H1134.1V61.92H1172.1V73.87H1134.1V88.14H1181.37V100H1122.24Z"
    fill="white"
  />
  <path
    d="M53.87 44.86H101.03V36.1L53.87 36.21C49.0305 36.2179 44.3917 38.1446 40.9706 41.5676C37.5494 44.9906 35.6253 49.6305 35.62 54.47V68.13H83C85.4896 68.1327 87.8764 69.1228 89.6368 70.8832C91.3972 72.6436 92.3874 75.0304 92.39 77.52V91.35H35.62V100H101V77.5C100.992 72.7221 99.0954 68.141 95.7235 64.756C92.3515 61.3709 87.7779 59.4564 83 59.43H44.29V54.43C44.2979 51.8926 45.3101 49.4615 47.1053 47.6682C48.9005 45.8749 51.3326 44.8653 53.87 44.86V44.86Z"
    fill="red"
  />
  <path
    d="M109.76 36.1V44.84H137.28V100H146.08V44.84H173.52V36.1H109.76Z"
    fill="red"
  />
  <path
    d="M258.13 91.2V36.16H249.33V100H313.09V91.2H258.13Z"
    fill="red"
  />
  <path
    d="M418.56 31.36L384.1 91.2H330.62V36.16H321.82V100H379.03H385.58H389.19L418.6 48.9L448.32 100.09H458.49L418.56 31.36Z"
    fill="red"
  />
  <path
    d="M240.59 44.84V36.1H182.26V100H240.59V91.2H191.06V68H221.76V59.3H191.06V44.84H240.59Z"
    fill="red"
  />
  <path
    d="M26.79 0.5C26.79 7.60976 23.9691 14.429 18.9464 19.4611C13.9238 24.4931 7.10974 27.3268 0 27.34C7.11841 27.34 13.9453 30.1678 18.9787 35.2013C24.0122 40.2347 26.84 47.0616 26.84 54.18C26.84 47.0616 29.6678 40.2347 34.7013 35.2013C39.7347 30.1678 46.5616 27.34 53.68 27.34C46.5616 27.34 39.7347 24.5122 34.7013 19.4788C29.6678 14.4453 26.84 7.61841 26.84 0.5H26.79Z"
    fill="white"
  />
  <path
    d="M438.63 47.28V38.49H433.57V36.1H445.65V38.49H441.16V47.28H438.63Z"
    fill="red"
  />
  <path
    d="M445.61 47.28V36.1H449.26L451.97 40.73L454.66 36.1H458.3V47.28H455.92V38L452 44.8L448 38V47.24L445.61 47.28Z"
    fill="red"
  />
  <title>
    STELLA Automotive AI
  </title>
</svg>
`;
